crates/matrix-sdk/src/client/builder.rs

```diff
@@ -233,13 +233,22 @@ impl ClientBuilder {
 
     /// Set up the store configuration for a SQLite store.
     #[cfg(feature = "sqlite")]
-    pub fn sqlite_store(
+    pub fn sqlite_store(self, path: impl AsRef<std::path::Path>, passphrase: Option<&str>) -> Self {
+        self.sqlite_store_with_cache_path(path, None::<&str>, passphrase)
+    }
+
+    /// Set up the store configuration for a SQLite store with an optional path
+    /// to separate cached data from state/crypto data.
+    #[cfg(feature = "sqlite")]
+    pub fn sqlite_store_with_cache_path(
         mut self,
         path: impl AsRef<std::path::Path>,
+        cache_path: Option<impl AsRef<std::path::Path>>,
```

Comment:
`Option<impl Into<PathBuf>>` seems a simpler approach, it removes the need to call `as_ref().to_owned()`, you just need to call `into()`. And it probably removes the need to get `None::<&str>`, simply having `None` would be fine.

Reply:
Ok so making the cache_path above non-optional all the way through to the new method on the SDK crate means that I didn't need to do this any more.

Reply:
(At least I think, its not the same as `path` which seems better for consistency?

Reply:
I think so.